### PR TITLE
Add pinch to zoom/zoom with mouse wheel

### PR DESCRIPTION
**Pull request by hverlin**
- add support for zooming using the mouse wheel (+ ctrl/cmd) or using the touchpad. Can be disabled if needed.
Screen Recording 2020-05-21 at 18 09 49
- fix scroll direction so that it is consistent with natural mouse scrolling

### CODE REVIEW

File: src/components/graph-view.js [hunk 1]
```diff
@@ -1698,6 +1698,21 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
       return;
     }
 
+    const isTouchPad = event.wheelDeltaY
+      ? event.wheelDeltaY !== -3 * event.deltaY
+      : event.deltaMode !== 0;
```
Comment (hodos) [inline src/components/graph-view.js]: is this a standard way to check for touchPad ?

File: src/components/graph-view.js [hunk 2]
```diff
@@ -1723,7 +1738,14 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
           deltaY: 0,
         };
 
-        this.setZoom(viewTransform.k || 1, offX, offY, 0);
+        if (isZooming) {
+          this.setDiscreteZoom(viewTransform.k * (1 + zoom), 0, [
```
Comment (hodos) [inline src/components/graph-view.js]: what's the difference between `setDiscreteZoom` and `setZoom`. Seems like they have a similar API and produce similar transforms 
cc @kbusemeyer

File: src/components/graph-view.js (hunk 1 above)
Comment (hverlin) [reply src/components/graph-view.js]: it's the only way I found on Chrome. In Firefox, it does not work, I need to look into it more.

File: src/components/graph-view.js (hunk 2 above)
Comment (hverlin) [reply src/components/graph-view.js]: `setDiscreteZoom` zoom on a point, vs `setZoom` translate and zoom. I would suggest to rename them if possible?

Comment (kbusemeyer) [reply src/components/graph-view.js]: ^ yup essentially what @hverlin said, for the new flow view, we just wanted to scale when zooming and not move to specific points and it was easier to just use the built in d3 methods. 

Also, the `modifyZoom`, which calls `setZoom` does some other calculations based off of previous transforms, but we knew the explicit zoom level we wanted to be at, so the new flow view just uses the `setDiscreteZoom` method as opposed to going through `modifyZoom`.

Feel free to change the naming, though!

File: src/components/graph-view.js (hunk 1 above)
Comment (hverlin) [reply src/components/graph-view.js]: simplified by always using the same constant

**Comment by hverlin** [comment]
@hodos @kbusemeyer it's ready to be merged. Also tried in the app and it works correctly.

**Comment by hodos** [comment]
I think we should hold off on landing this. Given that it doesn't work in FF and that @alexchantastic is working on zoom improvements

**Comment by hverlin** [comment]
it works with Firefox, you can try it

**Comment by hverlin** [comment]
at least you should definitely reverse the scrolling direction as it is not working correctly right now

**Comment by alexchantastic** [comment]
Can we pull the scroll direction fix into a different PR?

Pinch to zoom and mouse wheel zoom is on the list of enhancements we planned to make so it would be great if we could get this in. I will carve out some time to test later this week.

**Comment by hverlin** [comment]
it is guarded by a prop so we can merge and change later if something needs to be changed no?

**Review by hodos — APPROVED** [review]

**Pull request merged**